### PR TITLE
[data] Flag guard the dataset autoscaling

### PR DESCRIPTION
**Pull request by jianoaix**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/_internal/execution/streaming_executor_state.py [hunk 1]
```diff
@@ -424,17 +424,23 @@ def to_bundle(resource: ExecutionResources) -> Dict:
...
+            total_cpu_usage += task_bundle["CPU"] * op.num_active_work_refs()
         # Only include incremental resource usage for ops that are ready for
         # dispatch.
         if state.num_queued() > 0:
             # TODO(Clark): Scale up more aggressively by adding incremental resource
             # usage for more than one bundle in the queue for this op?
             resource_request.append(task_bundle)
 
+    total = ray.cluster_resources()
+    if not ("CPU" in total and total_cpu_usage > 0.6 * total["CPU"]):
```
Comment (ericl) [inline python/ray/data/_internal/execution/streaming_executor_state.py:442]: Hmm this shouldn't be needed right?

**Comment by c21** [comment]
Can we also add a backdoor environment variable to disable Data's autoscaler in this PR? It's been hard to debug Data vs Anyscale autoscaler. So allow it to be disabled would be easier for bisecting issue.

File: python/ray/data/_internal/execution/streaming_executor_state.py (hunk 1 above)
Comment (jianoaix) [reply python/ray/data/_internal/execution/streaming_executor_state.py:442]: Yeah we probably don't need to have this extra condition.

**Review by c21 — APPROVED** [review]

File: doc/source/data/dataset-internals.rst [hunk 2]
```diff
@@ -162,6 +162,7 @@ Configuring Resources and Locality
 By default, the CPU and GPU limits are set to the cluster size, and the object store memory limit conservatively to 1/4 of the total object store size to avoid the possibility of disk spilling.
 
 You may want to customize these limits in the following scenarios:
+
```
Comment (jianoaix) [inline doc/source/data/dataset-internals.rst:165]: Just a dop-by small format fix.

**Review by ericl — APPROVED** [review]

**Comment by jianoaix** [comment]
Test failures seem not relevant, but merge master and give it a try again.

**Pull request closed without merging**